### ISSUE TITLE
Duplicated errors when using a Literal choice

### ISSUE THREAD

**Issue by miketheman** [samuelcolvin/pydantic#768]
# Bug

Please complete:
* OS: macOS Mojave 10.14.6 (18G87)
* Python version `import sys; print(sys.version)`: 3.7.4 (default, Jul  9 2019, 18:13:23) [Clang 10.0.1 (clang-1001.0.46.4)]
* Pydantic version `import pydantic; print(pydantic.VERSION)`: 0.32.2

I tried looking for an existing issue, [found this](https://www.amazon.com/Low-Culture-Everything-Around-Computer/dp/B07381HM47), but didn't seem appropriate.

In #651, examples on how to use the `Literal` option were introduced, including [this example](https://github.com/samuelcolvin/pydantic/blob/bc600145184979d774e40c5d98f7a76d90ca616a/docs/examples/literal1.py) showing a single validation error, and the acceptable choices.

Running this file today produces two errors, instead of one, and doesn't populate the `permitted` value correctly.

```shellsession
$ python docs/examples/literal1.py
2 validation errors for Pie
flavor
  unexpected value; permitted: 'apple' (type=value_error.const; given=cherry; permitted=('apple',))
flavor
  unexpected value; permitted: 'pumpkin' (type=value_error.const; given=cherry; permitted=('pumpkin',))
```

**Comment by dmontagu** [comment]
Thanks for making an issue, I’ll take a look.

**Comment by dmontagu** [comment]
I found the source of the issue (Literal is converted to a union of literals of individual values; this was done to fix the generated schema). I'll see if I can't throw together an easy fix.

**Comment by samuelcolvin** [comment]
Fixed in #770, I'll deploy the patch in a few days to wait and see if anything else comes up.
